frontend/catalyst/jax_tracer.py

```diff
@@ -717,7 +717,12 @@ def apply_transform(transform_program, tape, flat_results):
 
     if is_program_transformed:
         is_valid_for_batch = is_transform_valid_for_batch_transforms(tape, flat_results)
-        tapes, post_processing = transform_program([tape])
+        tapes, post_processing_tuple_out = transform_program([tape])
+
+        def post_processing(r):
+            retval = post_processing_tuple_out(r)
+            return retval[0]
+
```

Comment:
You need something like:

```
        tapes, post_processing_tuple_out = transform_program([tape])

        if len(tape.measurements) == 1:
            def post_processing(r):
                retval = post_processing_tuple_out(r)
                return retval[0]
        else:
            post_processing = post_processing_tuple_out
```

Reply:
I'm not sure this is the best solution. The issue (I think) stems from something different being returned by the new device API and the old behaviour.

Reply:
In PennyLane Qnode we have that:

```
        res = qml.execute(
            (self._tape,),
            device=self.device,
            gradient_fn=self.gradient_fn,
            interface=self.interface,
            transform_program=full_transform_program,
            config=config,
            gradient_kwargs=self.gradient_kwargs,
            override_shots=override_shots,
            **self.execute_kwargs,
        )
        res = res[0]
 ```

Reply:
The difference with the new API is that no matter what we apply transforms from the device. Where before we did not.

Reply:
I did 'res = res[0]' for the new lightning and it seems to work.

Reply:
Thanks! I think this PR is essentially fixed by the work you did during the new device API. I will just add tests to make sure we don't have this regression.

Reply:
I think you are right :+1: